libraries/node-core-library/src/FileError.ts

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+import { FileLocationStyle, Path } from './Path';
+import { TypeUuid } from './TypeUuid';
+
+/**
+ * Provides options for the creation of a FileError.
+ *
+ * @public
+ */
+export interface IFileErrorOptions {
+  /**
+   * The absolute path to the file that contains the error.
+   */
+  absolutePath: string;
+
+  /**
+   * The root folder for the project that the error is in relation to.
+   */
+  projectFolder: string;
+
+  /**
+   * The line number of the error in the target file. Minimum value is 1.
+   */
+  line?: number;
+
+  /**
+   * The column number of the error in the target file. Minimum value is 1.
+   */
+  column?: number;
+}
+
+/**
+ * Provides options for the output message of a file error.
+ *
+ * @public
+ */
+export interface IFileErrorFormattingOptions {
+  /**
+   * The format for the error message. If no format is provided, format 'Unix' is used by default.
+   *
+   * @remarks If no format is specified and the environment variable TF_BUILD is set to 'True'
+   * (such as on Azure DevOps pipeline agents), the format 'AzureDevOps' is used by default.
```

Comment:
Didn't you remove the `AzureDevOps` format for now?

Reply:
Correct, forgot to remove this + the warning param.